Doc/whatsnew/3.8.rst

```diff
@@ -334,6 +334,9 @@ pathlib
 contain characters unrepresentable at the OS level.
 (Contributed by Serhiy Storchaka in :issue:`33721`.)
 
+Added :meth:`pathlib.Path.link()`.
```

Comment:
Should it be `pathlib.Path.link_to` ? 

Minor nit: Could add a short sentence that it creates a hardlink here and in NEWS entry. I will let it to reviewers on the wording though.